### PR TITLE
Add workflow parity plugins

### PR DESCRIPTION
**Pull request by benbalter**
This PR adds a four, enabled-by-default plugins that provide GitHub Pages with some additional parity with Markdown rendering on GitHub.com. Specifically:

* [Jekyll Optional Front Matter](https://github.com/benbalter/jekyll-optional-front-matter) - Don't require optional front matter to process a page as a page
* [Jekyll README Index](https://github.com/benbalter/jekyll-readme-index) - If a repo has a readme, but not an index, use the readme as the index
* [Jekyll Default Layout](https://github.com/benbalter/jekyll-default-layout) - Set a default per-page layout if no layout is specified and a logical one exists for the site
* [Jekyll Titles from Headings](https://github.com/benbalter/jekyll-titles-from-headings) - If there is no YAML title, use the first hading as the page's title

### CODE REVIEW

**Comment by benbalter** [comment]
In c8f4672 and 347b87f, I also added the Jekyll theme `jekyll-theme-primer` based on [Primer CSS](http://primercss.io/) as a default theme to provide minimal styling.

File: .rubocop.yml [hunk 1]
```diff
@@ -70,6 +70,7 @@ Metrics/ClassLength:
   Exclude:
     - !ruby/regexp /spec\/.*.rb$/
     - !ruby/regexp /features\/.*.rb$/
+    - lib/github-pages/configuration.rb
```
Comment (parkr) [inline .rubocop.yml]: It might be time to split some things out?

Comment (benbalter) [inline .rubocop.yml]: Broke things out via 2c739f6. Would love your :eyes:

File: lib/github-pages/configuration.rb [hunk 2]
```diff
@@ -2,43 +2,20 @@
...
-
-    # Plugins only allowed locally
-    DEVELOPMENT_PLUGINS = %w(
-      jekyll-admin
-    ).freeze
+    # Backward compatability of constants
+    DEFAULT_PLUGINS     = GitHubPages::Plugins::DEFAULT_PLUGINS
+    PLUGIN_WHITELIST    = GitHubPages::Plugins::PLUGIN_WHITELIST
+    DEVELOPMENT_PLUGINS = GitHubPages::Plugins::DEVELOPMENT_PLUGINS
+    THEMES              = GitHubPages::Plugins::THEMES
```
Comment (parkr) [inline lib/github-pages/configuration.rb]: 👍

**Pull request merged**